R/helpers.R

```diff
@@ -186,16 +186,39 @@ extract_num_iterations <- function(x) {
 # the model was trained with the `valids` parameter set such that error metrics
 # are saved for each tree on the model$record_evals attribute. The output
 # weights are useful for computing comps using leaf node assignments
-extract_tree_weights <- function(model, init_score, metric = "rmse") {
-  # Index into the errors list, and un-list so it is a flat/1dim list
-  record_evals <- model$record_evals
-  errors <- unlist(record_evals$tree_errors[[metric]]$eval)
-  errors <- c(init_score, errors)
-  diff_in_errors <- diff(errors, 1, 1)
+extract_tree_weights <- function(
+    model, init_score, training_data, outcome, metric) {
+  # Infer the number of trees in the model based on the error vector
+  record_evals <- model$record_evals$tree_errors[[metric]]$eval
+  num_iterations <- length(record_evals)
```

Comment:
Is there a better way to get the number of trees in a model? Based on `extract_num_iterations` above, it seems like maybe this is the best approach.

Reply:
Yes! It's a parameter stored in the model object, so something like `lgbm_final_full_fit$fit$params$num_iterations` will work.